akka-http2-support/src/main/scala/akka/http/impl/http2/WrappedSslContextSPI.scala

```diff
@@ -0,0 +1,83 @@
+package akka.http.impl.http2
+
+import java.security.SecureRandom
+import java.util
+import javax.net.ssl.KeyManager
+import javax.net.ssl.SSLContext
+import javax.net.ssl.SSLContextSpi
+import javax.net.ssl.SSLEngine
+import javax.net.ssl.SSLException
+import javax.net.ssl.SSLServerSocketFactory
+import javax.net.ssl.SSLSessionContext
+import javax.net.ssl.SSLSocketFactory
+import javax.net.ssl.TrustManager
+
+import akka.http.scaladsl.HttpsConnectionContext
+import org.eclipse.jetty.alpn.ALPN
+
+/**
+ * Wraps an SSLContext so that all engines it creates will announce HTTP/2 if possible.
+ */
+class WrappedSslContextSPI(underlying: SSLContext) extends SSLContextSpi {
```

Comment:
No, this is related to needing a change in akka-stream's TLS implementation, otherwise.